tests/scripts/thread-cert/border_router/MATN_16_LargeNumberOfMulticastGroupSubscriptionsToBBR.py

```diff
@@ -0,0 +1,231 @@
+        router1 = self.nodes[ROUTER_1]
+        host = self.nodes[HOST]
+
+        br1.set_backbone_router(reg_delay=REG_DELAY, mlr_timeout=consts.MLR_TIMEOUT_MIN)
+        br1.start()
+        self.simulator.go(5)
+        self.assertEqual('leader', br1.get_state())
+        self.assertTrue(br1.is_primary_backbone_router)
+
+        router1.start()
+        self.simulator.go(5)
+        self.assertEqual('router', router1.get_state())
+
+        host.start(start_radvd=False)
+        self.simulator.go(5)
+
+        # Loop steps 1 to 4 with i := 1 to 5.
+        for i in range(1, 6):
+            # 1. Router_1 registers for 15 multicast addresses, MASi.
+            router1.register_multicast_listener(*(MAS[i][1:]), timeout=consts.MLR_TIMEOUT_MIN)
+        self.simulator.go(5)
+
+        # Loop steps 5 to 6 with i := 1 to 5.
+        for i in range(1, 6):
+            # 5. Host sends a ping packet to the multicast address,
+            # MASi[ 3 * i - 1], on the backbone link.
+            self.assertFalse(
+                host.ping(MAS[i][i * 3 - 1],
+                          backbone=True,
+                          ttl=64,
+                          interface=host.get_ip6_address(config.ADDRESS_TYPE.ONLINK_ULA)[0]))
+
+            self.simulator.go(3)
+
+        # 7. Host multicasts a packet to the multicast addresses, MA2.
+        self.assertFalse(
+            host.ping(MA2, backbone=True, ttl=64, interface=host.get_ip6_address(config.ADDRESS_TYPE.ONLINK_ULA)[0]))
+        self.simulator.go(3)
+
+        self.collect_ipaddrs()
+        self.collect_rloc16s()
+        self.collect_rlocs()
+        self.collect_extra_vars()
+        self.collect_leader_aloc(BR_1)
+
+    def verify(self, pv: pktverify.packet_verifier.PacketVerifier):
+        pkts = pv.pkts
+        vars = pv.vars
+        pv.summary.show()
+
+        # Ensure the topology is formed correctly
+        pv.verify_attached('Router_1', 'BR_1')
+
+        # Loop steps 1 to 4 with i := 1 to 5.
+        for i in range(1, 6):
+            # 1. Router_1 registers for 15 multicast addresses, MASi.
+            # [Automatic result:]
+            # Unicasts an MLR.req CoAP request to BR_1 as follows:
+            # coap://[<BR_1 RLOC> or <PBBR ALOC>]:MM/n/mr
+            # Where the payload contains:
+            # IPv6 Addresses TLV: MASi (15 addresses)
+            _pkt = pkts.filter_wpan_src64(vars['Router_1']) \
+                .filter_ipv6_2dsts(vars['BR_1_RLOC'], consts.PBBR_ALOC) \
+                .filter_coap_request('/n/mr') \
+                .filter(lambda p: set(p.thread_meshcop.tlv.ipv6_addr) == set(MAS[i][1:])) \
+                .must_next()
+
+            # 2. BR_1 Responds to the multicast registration.
+            # BR_1 unicasts an MLR.rsp CoAP response to Router_1 as follows:
+            # 2.04 changed
+            # Where the payload contains:
+            # Status TLV: ST_MLR_SUCCESS
+            pkts.copy().filter_wpan_src64(vars['BR_1']) \
```

Comment:
Is `copy()` necessary?

```suggestion
            pkts.filter_wpan_src64(vars['BR_1']) \
```

`pktverify` should be able to handle adjacent WPAN and Ethernet packets correctly even when they are disordered.